iothub/device/src/Fx/Fx.cs

```diff
@@ -10,14 +10,6 @@ namespace Microsoft.Azure.Devices.Client
 {
     internal static class Fx
     {
-        public static void AssertAndThrow(bool condition, string description)
-        {
-            if (!condition)
-            {
-                throw new InvalidOperationException(description);
-            }
-        }
-
         public static bool IsFatal(Exception ex)
```

Comment:
Can we move this method to the exception class as an internal/private static too?

Reply:
We can if we want to, but the codes calling it will be something like 
```c#
catch (Exception ex) when (!IotHubClientException.IsFatal(ex))
{
    ......
}
```
kindof weird IMO. 

I would like to keep this method within a separate file but maybe we can move this file into the "Exceptions" folder for the hub device and service packages.

Reply:
Does `ex.IsTransient == !IsFatal(ex)`?

If so, can we simply use ex.IsTransient instead?

Reply:
Oh, a fatal exception may not always be IotHubClientException, right? I forgot about that.

Reply:
Yes, that's what I am worried about.